File: lib/models/vehicle_model.dart
class VehicleModel {
  final int vehicleId;
  final String? vehicleType;
  final String? plateNumber;
  final int? capacity;
  final String? status;
  final DateTime createdAt;
  final String? driverName;
  final String? driverContact;
  final String? barangayName;
  final String? landmark;
  final double? latitude;
  final double? longitude;
  final int? currentOccupants;

  VehicleModel({
    required this.vehicleId,
    this.vehicleType,
    this.plateNumber,
    this.capacity,
    this.status,
    required this.createdAt,
    this.driverName,
    this.driverContact,
    this.barangayName,
    this.landmark,
    this.latitude,
    this.longitude,
    this.currentOccupants,
  });

  factory VehicleModel.fromJson(Map<String, dynamic> json) {
    return VehicleModel(
      vehicleId: json['vehicle_id'] ?? 0,
      vehicleType: json['vehicle_type'],
      plateNumber: json['plate_number'],
      capacity: json['capacity'],
      status: json['status'],
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
      driverName: json['driver_name'],
      driverContact: json['driver_contact'],
      barangayName: json['barangay_name'],
      landmark: json['landmark'],
      latitude: json['latitude'] != null
          ? double.parse(json['latitude'].toString())
          : null,
      longitude: json['longitude'] != null
          ? double.parse(json['longitude'].toString())
          : null,
      currentOccupants: json['current_occupants'],
    );
  }

  Map<String, dynamic> toJson() => {
    'vehicle_id': vehicleId,
    'vehicle_type': vehicleType,
    'plate_number': plateNumber,
    'capacity': capacity,
    'status': status,
    'created_at': createdAt.toIso8601String(),
    'driver_name': driverName,
    'driver_contact': driverContact,
    'barangay_name': barangayName,
    'landmark': landmark,
    'latitude': latitude,
    'longitude': longitude,
    'current_occupants': currentOccupants,
  };
}

File: Evacuways_Admin/app/models/VehicleModel.php
<?php
class VehicleModel {
    private $conn;
    private $table = "evacuways_vehicles";

    public function __construct($db) {
        $this->conn = $db;
    }

    public function create($type, $plate, $capacity, $barangay_name, $landmark, $status, $driver_name, $driver_contact) {
        $query = "INSERT INTO " . $this->table . " 
                  (vehicle_type, plate_number, capacity, barangay_name, landmark, status, driver_name, driver_contact) 
                  VALUES (:type, :plate, :capacity, :barangay_name, :landmark, :status, :driver_name, :driver_contact)";
        
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([
            ':type' => $type,
            ':plate' => $plate,
            ':capacity' => $capacity,
            ':barangay_name' => $barangay_name,
            ':landmark' => $landmark,
            ':status' => $status,
            ':driver_name' => $driver_name,
            ':driver_contact' => $driver_contact
        ]);
    }

    public function update($id, $type, $plate, $capacity, $barangay_name, $landmark, $status, $driver_name, $driver_contact) {
        $query = "UPDATE " . $this->table . " 
                  SET vehicle_type = :type, 
                      plate_number = :plate, 
                      capacity = :capacity, 
                      barangay_name = :barangay_name, 
                      landmark = :landmark,
                      status = :status,
                      driver_name = :driver_name,
                      driver_contact = :driver_contact
                  WHERE vehicle_id = :id";
        
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([
            ':id' => $id,
            ':type' => $type,
            ':plate' => $plate,
            ':capacity' => $capacity,
            ':barangay_name' => $barangay_name,
            ':landmark' => $landmark,
            ':status' => $status,
            ':driver_name' => $driver_name,
            ':driver_contact' => $driver_contact
        ]);
    }

    public function getAll() {
        $query = "SELECT * FROM " . $this->table . " ORDER BY created_at DESC, vehicle_id DESC";
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public function getStatusStats() {
        $query = "SELECT status, COUNT(*) as count FROM " . $this->table . " GROUP BY status";
        $stmt = $this->conn->prepare($query);
        $stmt->execute();
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    public function delete($id) {
        $query = "DELETE FROM " . $this->table . " WHERE vehicle_id = :id";
        $stmt = $this->conn->prepare($query);
        return $stmt->execute([':id' => $id]);
    }
}